frontend/src/lib/components/LemonCheckbox/LemonCheckbox.tsx

```diff
@@ -0,0 +1,78 @@
+import clsx from 'clsx'
+import React, { useEffect, useMemo, useState } from 'react'
+import './LemonCheckbox.scss'
+
+export interface LemonCheckboxProps {
+    checked?: boolean
+    defaultChecked?: boolean
+    indeterminate?: boolean
+    disabled?: boolean
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
+    label?: string | JSX.Element
+    id?: string
+    className?: string
+    style?: React.CSSProperties
+    color?: string
+}
+
+export function LemonCheckbox({
+    checked,
+    defaultChecked,
+    indeterminate,
+    disabled,
+    onChange,
+    label,
+    id,
+    className,
+    style,
+    color,
+}: LemonCheckboxProps): JSX.Element {
+    const [localChecked, setLocalChecked] = useState(checked ?? defaultChecked)
+
+    useEffect(() => {
+        setLocalChecked(checked)
+    }, [checked])
+
+    id = useMemo(() => id || `lemon-checkbox-${Math.floor(Math.random() * 1000000)}`, [id])
```

Comment:
Any reason we're not creating a new variable id immutably here?

Reply:
What do you mean by "immutably"?

Reply:
I guess Alex means replacing:

```tsx
function Component() {
    const id = useMemo(() => rawId || `lemon-checkbox-${Math.floor(Math.random() * 1000000)}`, [rawId])
}
```

with

```tsx
let i = 0
function Component() {
    const id = useMemo(() => rawId || `lemon-checkbox-${i++}`, [rawId])
}
```

Although small, there's a legit chance someone somewhere will eventually have two checkboxes that get their values synced because of the same random ID, leading to some weirdness.

Reply:
I see, so basically "sequential". Made it so.